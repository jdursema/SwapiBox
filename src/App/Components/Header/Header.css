h1{
  color: #FFE400;
  font-size: 100px;
  font-family: 'Pathway Gothic One', sans-serif;
  margin: 0;
  margin-bottom: 5%;
  line-height: 10px;
}

img{
  width: 10%;
  display: flex;
  justify-content: right;
  margin-bottom: 0;
  margin-left: 32%;
}

.favorite-btn{
  background-color: skyblue;
  font-size: 20px;
  width: 150px;
  border-radius: 0;
  border: none;
  margin: 1%;
  padding: 1% 0;
}

.btn-container{
  display: flex;
  justify-content: flex-end;
  margin: 0;
}