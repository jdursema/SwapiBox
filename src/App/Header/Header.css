h1{
  color: yellow;
  font-size: 50px;
  margin: 0;
}

.favorite-btn{
  background-color: skyblue;
  font-size: 20px;
  width: 150px;
  border-radius: 0;
  border: none;
  margin: 1% 0;
  padding: 1% 0;
}

.btn-container{
  display: flex;
  justify-content: flex-end;
  margin: 0;
}