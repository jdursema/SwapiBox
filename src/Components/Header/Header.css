.favorite-btn{
  background-color: #f3a633;
  color: #FEFEFE;
  font-family: 'Pathway Gothic One', sans-serif;
  font-size: 30px;
  width: 175px;
  border-radius: 0;
  border: none;
  margin: 1%;
  padding: 1% 0;
}

.btn-container{
  display: flex;
  justify-content: flex-end;
  margin: 0;
}