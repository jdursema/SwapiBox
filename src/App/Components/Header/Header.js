import React from 'react';
import './Header.css';
import PropTypes from 'prop-types';

const Header = ({ Favorites, displayFavorites }) => {
  return (
    <div className='header'>
      <div className='btn-container'>
        <button onClick={()=>displayFavorites()} className='favorite-btn'>
          Favorites {Favorites.length}
        </button>
      </div>
      <img src='https://c.tribune.com.pk/2015/03/853137-starwars-1426319002-513-640x480.jpg'/>
      <h1>Return of the Swapi Box</h1>
    </div>
  );
};

export default Header;

Header.propTypes = {
  Favorites: PropTypes.array,
  displayFavorites: PropTypes.func
};