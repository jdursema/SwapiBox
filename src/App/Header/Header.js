import React from 'react';
import './Header.css';
import PropTypes from 'prop-types';

const Header = ({Favorites, displayFavorites}) => {
  return (
    <div className='header'>
      <div className='btn-container'>
        <button 
          onClick={()=>displayFavorites()} className='favorite-btn'>Favorites {Favorites.length}</button>
      </div>
      <h1>Return of the SwapiBox</h1>
    </div>
  );
};

export default Header;

Header.propTypes = {
  Favorites: PropTypes.array,
  displayFavorites: PropTypes.func
};