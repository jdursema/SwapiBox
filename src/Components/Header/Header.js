import React from 'react';
import './Header.css';
import PropTypes from 'prop-types';

const Header = ({ Favorites, displayFavorites }) => {
  return (
    <div className='header'>
        <h1 className='title'>SW(api) Box</h1>
        <button onClick={()=>displayFavorites()} className='favorite-btn'>
          Favorites {Favorites.length}
        </button>
    </div>
  );
};

export default Header;

Header.propTypes = {
  Favorites: PropTypes.array,
  displayFavorites: PropTypes.func
};