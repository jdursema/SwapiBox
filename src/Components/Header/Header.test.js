import React from 'react';
import { shallow } from 'enzyme';
import Header from './Header';

it('renders without crashing', () => {
  const mockFunc = jest.fn();
  const favorites =
    [{name: 'Julie', Data:{Homeworld: 'earth'}}, 
      {name: 'Pluto', Data:{Residents: 'Goofy'}}];
  const header = shallow(<Header Favorites={favorites} displayFavorites={mockFunc}/>);

  expect(header).toBeDefined();
});

it('should match snap shot', () => {
  const mockFunc = jest.fn();
  const favorites =
    [{name: 'Julie', Data:{Homeworld: 'earth'}}, 
      {name: 'Pluto', Data:{Residents: 'Goofy'}}];
  const header = shallow(<Header Favorites={favorites} displayFavorites={mockFunc}/>);

  expect(header).toMatchSnapshot();
});


it('should call the display favorites function after the favorite button has been clicked', () =>{
  const mockFunc = jest.fn();
  const favorites =
    [{name: 'Julie', Data:{Homeworld: 'earth'}}, 
      {name: 'Pluto', Data:{Residents: 'Goofy'}}];
  const header = shallow(<Header Favorites={favorites} displayFavorites={mockFunc}/>);
  const button = header.find('button');

  expect(mockFunc.mock.calls.length).toEqual(0);
  
  button.simulate('click');
  
  expect(mockFunc.mock.calls.length).toEqual(1);
});