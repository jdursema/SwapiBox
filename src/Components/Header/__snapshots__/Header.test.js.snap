// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snap shot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header
    Favorites={
        Array [
            Object {
              "Data": Object {
                "Homeworld": "earth",
              },
              "name": "Julie",
            },
            Object {
              "Data": Object {
                "Residents": "Goofy",
              },
              "name": "Pluto",
            },
          ]
    }
    displayFavorites={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="btn-container"
>
          <button
                    className="favorite-btn"
                    onClick={[Function]}
          >
                    Favorites 
                    2
          </button>
</div>,
        <img
          src="https://c.tribune.com.pk/2015/03/853137-starwars-1426319002-513-640x480.jpg"
/>,
        <h1>
          Return of the Swapi Box
</h1>,
      ],
      "className": "header",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <button
            className="favorite-btn"
            onClick={[Function]}
>
            Favorites 
            2
</button>,
          "className": "btn-container",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Favorites ",
              2,
            ],
            "className": "favorite-btn",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Array [
            "Favorites ",
            2,
          ],
          "type": "button",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "src": "https://c.tribune.com.pk/2015/03/853137-starwars-1426319002-513-640x480.jpg",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Return of the Swapi Box",
        },
        "ref": null,
        "rendered": "Return of the Swapi Box",
        "type": "h1",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="btn-container"
>
            <button
                        className="favorite-btn"
                        onClick={[Function]}
            >
                        Favorites 
                        2
            </button>
</div>,
          <img
            src="https://c.tribune.com.pk/2015/03/853137-starwars-1426319002-513-640x480.jpg"
/>,
          <h1>
            Return of the Swapi Box
</h1>,
        ],
        "className": "header",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <button
              className="favorite-btn"
              onClick={[Function]}
>
              Favorites 
              2
</button>,
            "className": "btn-container",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Favorites ",
                2,
              ],
              "className": "favorite-btn",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Array [
              "Favorites ",
              2,
            ],
            "type": "button",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "src": "https://c.tribune.com.pk/2015/03/853137-starwars-1426319002-513-640x480.jpg",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Return of the Swapi Box",
          },
          "ref": null,
          "rendered": "Return of the Swapi Box",
          "type": "h1",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
